lib/liveman/user/schemas/user_schema.ex

```diff
@@ -3,15 +3,49 @@ defmodule Liveman.User.Schemas.User do
 
   import Ecto.Changeset
 
+  @email_regex_format ~r/^\S+@\S+\.\S+$/
+
+  @derive {Inspect, except: [:password]}
+  @primary_key {:id, Ecto.UUID, autogenerate: true}
   schema "users" do
     field(:email, :string)
+    field(:password, :string, virtual: true)
+    field(:hashed_password, :string)
 
     timestamps()
   end
 
   def changeset(individual \\ %__MODULE__{}, attrs) do
     individual
-    |> cast(attrs, [:email])
+    |> cast(attrs, [:email, :password])
+    |> validate_email()
+    |> validate_password()
+  end
+
+  defp validate_email(changeset) do
+    changeset
     |> validate_required([:email])
+    |> validate_format(:email, @email_regex_format, message: "is invalid")
+    |> validate_length(:email, max: 160)
+    |> unique_constraint(:email)
+  end
+
+  defp validate_password(changeset) do
+    changeset
+    |> validate_required([:password])
+    |> validate_length(:password, min: 6, max: 80)
+    |> update_hashed_password
```

Comment:
Minor naming preference for:

```suggestion
    |> put_hashed_password
```

It is more idiomatic as it actually puts a change into a map 💭